rosenpass/src/protocol.rs

```diff
@@ -548,7 +548,7 @@ impl CryptoServer {
     pub fn pidm(&self) -> Result<PeerId> {
         Ok(Public::new(
             hash_domains::peerid()?
-                .mix(self.spkm.secret())?
+                .mix(&*self.spkm)?
```

Comment:
Looks all good…how about using .deref() or .borrow() here (as appropriate) to avoid this magic looking incantation?